### DISCUSSION TITLE
Disable cursor highlighting?

### DISCUSSION THREAD

**Discussion by poperigby** [lukas-reineke/indent-blankline.nvim#165] [Q&A]
Love the plugin, but I don't like how it makes my cursor grey when I hover over the indented whitespace. In the help page, I see quite a few highlight groups, but all of the ones that look like they might be the right one say they use the whitespace highlight group by default.

**Comment by lukas-reineke** [comment]
AFAIK the underlying highlight group can't change how the cursor is displayed.
You only have `:help hl-Cursor` highlight group for the cursor. And it's the same everywhere.

Some terminal emulators also overwrite how the cursor is displayed, you'll have to look at the documentation for your one.

**Reply by poperigby** [reply]
Oh yeah, you're right about the cursor. I think it might have something to do with the color for the indented whitespace being grey, while the other whitespace is translucent or something.

Indent whitespace:
![swappy-20210805_162826](https://user-images.githubusercontent.com/20866468/128434043-ae3f1166-ed3b-4ea5-9171-d0377634bfeb.png)

Other whitespace:
![swappy-20210805_162839](https://user-images.githubusercontent.com/20866468/128434055-a0601484-5edc-460a-97f8-188db6ef7243.png)

**Comment by Congee** [comment]
> AFAIK the underlying highlight group can't change how the cursor is displayed.

You can change the color of your cursor. You just need to setup `guicursor` using the example 
```
:set guicursor=n-v-c:block,i-ci-ve:ver25,r-cr:hor20,o:hor50
  \,a:blinkwait700-blinkoff400-blinkon250-Cursor/lCursor  <---------- this is the key
  \,sm:block-blinkwait175-blinkoff150-blinkon175
```
And, importantly, `highlight Cursor gui=nocombine`

**Reply by lukas-reineke** [reply]
Yes you can change the cursor highlight directly.
But the highlight group under the cursor doesn't affect the cursor.

**Reply by Congee** [reply]
ummm, what does that mean? Aren't I changing the cursor by the highlight group `Cursor`?

**Reply by lukas-reineke** [reply]
Yes, and the plugin does not touch `Cursor`.
The original question was _to not change_ the cursor highlight. Which this plugin does not do.
Highlight groups of something under the cursor do not influence the cursor highlight group.

You can make your cursor be the inverse of the color underneath, but this is not something I control in this plugin.

**Reply by Congee** [reply]
Please pardon me if I ask a stupid question again
> Highlight groups of something under the cursor do not influence the cursor highlight group

I still have the same question about whether this is true or not. As far as I can tell, it's hard to have any effect by doing `hi Cursor ....`, but as long as you override the original highlight settings of the group Cursor `hi Cursor gui=nocombine` (note that neither `hi clear Cursor` nor `hi Cursor gui=NONE` work), you'll be able to freely customize the Cursor group like `hi Cursor guifg=red`.

Per my understanding, what you meant is that it is impossible to alter the highlights of group Cursor. Did you mean it?

**Reply by Congee** [reply]
BTW, lack of contrast in the indented area, especially with a dark colorscheme, is the only problem I have with `lukas-reineke/indent-blankline.nvim`. Now it's solved using the above hacks. This plugin is truly awesome :+1:

**Reply by lukas-reineke** [reply]
> Per my understanding, what you meant is that it is impossible to alter the highlights of group Cursor. Did you mean it?

No, what I mean is it is impossible to change the highlight of the cursor with the highlight group of whatever is under the cursor.
If your cursor is over a string. Nothing you do to the `String` highlight group will change how the cursor is highlighted.
